caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/AdResolver.java

```diff
@@ -122,6 +111,11 @@ public URL toURL(URI uri) {
 
         try {
             String path = getPath(uri);
+            
+            // check if authMethod has been set
+            if (this.authMethod == null) {
+                this.authMethod = AuthenticationUtil.getAuthMethod(AuthenticationUtil.getCurrentSubject());
```

Comment:
Assigning the AuthMethod from the current subject to the member variable will make re-use of the resolver buggy; it should be assigned to the local var (am) instead.